server/src/main/java/org/apache/kafka/server/config/AbstractKafkaConfig.java

```diff
@@ -161,6 +169,76 @@ public static Map<String, String> getMap(String propName, String propValue) {
         }
     }
 
+    public static List<Endpoint> listenerListToEndPoints(List<String> listeners, Map<ListenerName, SecurityProtocol> securityProtocolMap) {
+        return listenerListToEndPoints(listeners, securityProtocolMap, true);
+    }
+
+    public static List<Endpoint> listenerListToEndPoints(List<String> listeners, Map<ListenerName, SecurityProtocol> securityProtocolMap, boolean requireDistinctPorts) {
+        List<Endpoint> endPoints;
+        try {
+            endPoints = SocketServerConfigs.listenerListToEndPoints(listeners, securityProtocolMap);
+        } catch (Exception e) {
+            throw new IllegalArgumentException(String.format("Error creating broker listeners from '%s': %s", listeners, e.getMessage()), e);
+        }
+        validate(endPoints, listeners, requireDistinctPorts);
+        return endPoints;
+    }
```

Comment:
Could we do the following?
```suggestion
    public static List<Endpoint> listenerListToEndPoints(List<String> listeners, Map<ListenerName, SecurityProtocol> securityProtocolMap, boolean requireDistinctPorts) {
        try {
            List<Endpoint> endPoints = SocketServerConfigs.listenerListToEndPoints(listeners, securityProtocolMap);
            validate(endPoints, listeners, requireDistinctPorts);
            return endPoints;
        } catch (Exception e) {
            throw new IllegalArgumentException(String.format("Error creating broker listeners from '%s': %s", listeners, e.getMessage()), e);
        }
    }
```